clang/test/lab1/kozyreva_ekaterina/tests.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/kozyrevaPrintNamesPlugin%pluginext -plugin classprinter %s 1>&1 | FileCheck %s
+
+
+class Test1 {
+  int x;
+  double y;
+};
+// CHECK: Test1
+// CHECK-NEXT: |_x
+// CHECK-NEXT: |_y
+
+struct Test2 {
+  float a;
+  char b;
+  bool c;
+};
+// CHECK: Test2
+// CHECK-NEXT: |_a
+// CHECK-NEXT: |_b
+// CHECK-NEXT: |_c
+
+template <typename T> class Test3 { T var; };
+// CHECK: Test3
+// CHECK-NEXT: |_var
+
+class Test4 {
+  class Nested {
+    int z;
+  };
+};
+// CHECK: Test4
+// CHECK: Nested
+// CHECK-NEXT: |_z
```

Comment:
Please add a test for static member fields also.

Reply:
Still static members printing is required.

Reply:
I'm sorry, I've tried everything. When I add a test for static fields, it doesn't work. Tell me, is there a bug in the plugin somewhere or am I writing tests wrong somehow? I can't think of anything anymore, I haven't found anything.

Reply:
Static fields of a class are saved in a bit different nodes of AST tree. That is why your plugin doesn't print them now. You may ask your groupmates who already made your variant regarding printing static fields.

Reply:
Static fields comment is not resolved @E-Kozyreva